### ISSUE TITLE
mock-fs does not work as expected for Node >20.4

### ISSUE THREAD

**Issue by stennie** [basti1302/repoman#61]
`mock-fs` is currently used for `test/generator/local.js`. Tests pass in Node 20.4 and earlier, but fail in more recent versions of Node.

Possible related to: https://github.com/tschaub/mock-fs/issues/377

Since this is a very limited use case, I think it could be resolved by either:
 - creating a sample directory to avoid the need for monkey patching core `fs` behaviour
 -  using `memfs` or an alternative package

**Comment by basti1302** [comment]
I totally agree. The sample directory sounds like the most straightforward option to me.
